src/fsipy/automatedPostprocessing/postprocessing_h5py/create_spectrograms_chromagrams.py

```diff
@@ -0,0 +1,256 @@
+        output_amplitudes = output_amplitudes[output_amplitudes[:, 0] <= end_t]
+        ax5.plot(output_amplitudes[:, 0], output_amplitudes[:, 10], label=case_name)
+        ax5.set_ylabel("Amplitude")
+        ax5.set_xlabel('Time (s)')
+    else:
+        ax4.set_xlabel('Time (s)')
+
+    # Name composite figure and save
+    composite_figure_name = f"{quantity}_{case_name}_{num_windows}_windows_thresh{min_color}_composite_figure"
+    if power_scaled:
+        composite_figure_name += "_power_scaled"
+
+    composite_figure_path = Path(image_folder) / (composite_figure_name + '.png')
+    logging.info(f"\n--- Saving composite figure to {composite_figure_path}\n")
+    fig1.savefig(composite_figure_path)
+
+    # create separate spectrogram figure
+    fig2, ax2_1 = plt.subplots()
+    fig2.set_size_inches(7.5, 5)
+    title = f"Pxx max = {max_val:.2e}, Pxx min = {min_val:.2e}, threshold Pxx = {lower_thresh}"
+    spec.plot_spectrogram(fig2, ax2_1, bins, freqs, Pxx, ylim, title=title,
+                          x_label="Time (s)", color_range=[min_color, max_color])
+
+    # Save data to files (spectrogram)
+    spec_filename = f"{quantity}_{case_name}_{num_windows}_windows_thresh{min_color}_spectrogram"
+    path_to_spec = Path(image_folder) / (spec_filename + '.png')
+    logging.info(f"--- Saving spectrogram figure to {path_to_spec}\n")
+    fig2.savefig(path_to_spec)
+
+    output_csv_path = path_to_spec.with_suffix('.csv')
+    data_csv = np.append(freqs[np.newaxis].T, Pxx, axis=1)
+    bins_txt = np.array2string(bins, max_line_width=10000, precision=2, separator=',').replace("[", "").replace("]", "")
+    np.savetxt(output_csv_path, data_csv, header=bins_txt, delimiter=",")
+
+    # Save data to files (chromagram)
+    chroma_filename = f"{quantity}_{case_name}_{num_windows}_windows_chromagram"
+    path_to_chroma = Path(image_folder) / (chroma_filename + '.png')
+    chroma_y = np.linspace(0, 1, chroma.shape[0])
+
+    output_csv_path = path_to_chroma.with_suffix('.csv')
+    data_csv = np.append(chroma_y[np.newaxis].T, chroma, axis=1)
+    bins_txt = \
+        np.array2string(bins_raw, max_line_width=10000, precision=2, separator=',').replace("[", "").replace("]", "")
+    np.savetxt(output_csv_path, data_csv, header=bins_txt, delimiter=",")
+
+    # Save data to files (SBI)
+    sbi_filename = f"{quantity}_{case_name}_{num_windows}_windows_SBI"
+    path_to_sbi = Path(image_folder) / (sbi_filename + '.png')
+
+    csv_output_path = path_to_sbi.with_suffix('.csv')
+    data_csv = np.array([bins, chroma_entropy]).T
+    np.savetxt(csv_output_path, data_csv, header="t (s), SBI", delimiter=",")
+
+
+def main():
+    # Load in case-specific parameters
+    args = spec.read_command_line_spec()
+
+    # Create logger and set log level
+    logging.basicConfig(level=args.log_level, format="%(message)s")
+
+    # Load parameters from default_parameters.json
+    parameters = read_parameters_from_file(args.folder)
```

Comment:
Is it possible to add `assert` to check if the `args.folder` exist?

Reply:
I had wrong folder path but the error message came from not finding `.json` file so it is my stupid mistake but would be nice to check if the folder exist in the first place.